api/graphql/schema.graphql

```diff
@@ -148,6 +151,35 @@ type User {
     @canResolved(ability: "view") # Search requests submitted by a user
 }
 
+type EmployeeProfile {
+  organizationTypeInterest: [LocalizedOrganizationTypeInterest!]
+    @rename(attribute: "career_planning_organization_type_interest")
+  moveInterest: [LocalizedMoveInterest!]
+    @rename(attribute: "career_planning_move_interest")
+  mentorshipStatus: [LocalizedMentorship!]
+    @rename(attribute: "career_planning_mentorship_status")
+  mentorshipInterest: [LocalizedMentorship!]
+    @rename(attribute: "career_planning_mentorship_interest")
+  execInterest: Boolean @rename(attribute: "career_planning_exec_interest")
+  execCoachingStatus: [LocalizedExecCoaching!]
+    @rename(attribute: "career_planning_exec_coaching_status")
+  execCoachingInterest: [LocalizedExecCoaching!]
+    @rename(attribute: "career_planning_exec_coaching_interest")
+  aboutYou: String @rename(attribute: "career_planning_about_you")
+  careerGoals: String @rename(attribute: "career_planning_career_goals")
+  learningGoals: String @rename(attribute: "career_planning_learning_goals")
+  workStyle: String @rename(attribute: "career_planning_work_style")
+  dreamRoleTitle: String @rename(attribute: "dream_role_title")
+  dreamRoleAdditionalInformation: String
+    @rename(attribute: "dream_role_additional_information")
+  dreamRoleCommunity: Community @belongsTo
+  dreamRoleClassification: Classification @belongsTo
+  dreamRoleWorkStream: WorkStream @belongsTo
+  dreamRoleDepartments: [Department!] @belongsToMany
+
+  userPublicProfile: UserPublicProfile
```

Comment:
The [issue](https://github.com/GCTC-NTGC/gc-digital-talent/issues/12001#:~:text=It%20should%20be%20protected%20by%20%40canResolved(ability%3A%20%22view%22)%2C%20because%20some%20managers%20who%20may%20have%20permission%20to%20view%20the%20applicant%20profile%20should%20not%20be%20able%20to%20see%20the%20employee%20profile.) states: `userPublicProfile: UserPublicProfile @self`.

Reply:
Does it work without self? Cause it should, `@self` is when you want to reference the root model which this does not 🤔

Reply:
I agree this field is about relationship between EmployeeProfile and UserPublicProfile, which are also different types.